forge-game/src/main/java/forge/game/ability/effects/DestroyAllEffect.java

```diff
@@ -76,7 +79,10 @@ public void resolve(SpellAbility sa) {
         if (sa.hasParam("RememberAllObjects")) {
             card.addRemembered(list);
         }
-
+        
+        if (isOptional && !sa.getActivatingPlayer().getController().confirmAction(sa, null, Localizer.getInstance().getMessage("lblWouldYouLikeDestroy",(desc)), null)) {;
```

Comment:
you can probably add another if above here that returns if `list.isEmpty()`
that way you avoid getting the prompt if target is unequipped

Reply:
I've tested it and it seems to work, no prompt if no equipment or is something still missing ?